docs/architecture-diagrams.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
+"""
+Generate architecture diagrams for the documentation
+"""
+from diagrams import Diagram, Edge, Cluster
+from diagrams.k8s.compute import Deployment, Pod
+from diagrams.k8s.network import Service
+from diagrams.k8s.storage import PVC
+
+IMAGES_DIR = "source/_static/images"
+
+with Diagram("JupyterHub + SSH + SFTP", show=False, outformat="png", filename=f"{IMAGES_DIR}/architecture") as diag:
+    autohttps = Deployment("autohttps")
+    ssh = autohttps >> Edge(label="port 22") >> Deployment("jupyterhub-ssh")
+    sftp = autohttps >> Edge(label="port 2222") >> Deployment("jupyterhub-sftp")
```

Comment:
1. The SFTP k8s service expose a port with name ssh and number 22 (Recent change by me from number 2222)
2. The SFTP k8s service targets a port with the name ssh on the pod (Recent change by me from targeting port 2222 by number rather than name)
3. The SFTP k8s pod expose a port with name ssh and number 2222

Services can have sensible numbers <1000, pods should stay above 1000 though to comply with pod security policies etc.

```suggestion
    sftp = autohttps >> Edge(label="port 22") >> Deployment("jupyterhub-sftp")
```